From 80309d7c7be8f8f1444a845d3ffdeefcda89b54c Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Wed, 5 Sep 2018 16:49:45 +0530
Subject: [PATCH 125/909] clk: renesas: rcar-gen3: Add division value argument
 to cpg_zg_clk_register

commit 4abe3c97812b6c90cd54aa19a0ece77dbcb1e25a from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This patch adds an argument to cpg_zg_clk_register() that sets the division
value of the zg clock divider in struct cpg_core_clk[].

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/renesas/rcar-gen3-cpg.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/clk/renesas/rcar-gen3-cpg.c b/drivers/clk/renesas/rcar-gen3-cpg.c
index 8130b630fc14..89937e7c57af 100644
--- a/drivers/clk/renesas/rcar-gen3-cpg.c
+++ b/drivers/clk/renesas/rcar-gen3-cpg.c
@@ -356,7 +356,8 @@ static struct clk * __init cpg_z_clk_register(const char *name,
 
 static struct clk * __init cpg_zg_clk_register(const char *name,
 					       const char *parent_name,
-					       void __iomem *reg)
+					       void __iomem *reg,
+					       unsigned int div)
 {
 	struct clk_init_data init;
 	struct cpg_z_clk *zclk;
@@ -376,7 +377,7 @@ static struct clk * __init cpg_zg_clk_register(const char *name,
 	zclk->kick_reg = reg + CPG_FRQCRB;
 	zclk->hw.init = &init;
 	zclk->mask = CPG_FRQCRB_ZGFC_MASK;
-	zclk->fixed_div = 4; /* PLLVCO x 1/2 x 3DGE divider x 1/2 */
+	zclk->fixed_div = div; /* PLLVCO x 1/div1 x 3DGE divider x 1/div2 */
 
 	clk = clk_register(NULL, &zclk->hw);
 	if (IS_ERR(clk))
@@ -757,7 +758,7 @@ struct clk * __init rcar_gen3_cpg_clk_register(struct device *dev,
 
 	case CLK_TYPE_GEN3_ZG:
 		return cpg_zg_clk_register(core->name, __clk_get_name(parent),
-					   base);
+					   base, core->div);
 
 	default:
 		return ERR_PTR(-EINVAL);
-- 
2.17.1

